Install default config under `/usr/share/doc/openqa/examples` (#6347)
os-autoinst/openQA: docs/GettingStarted.asciidoc

you must pass it to `testapi::assert_and_click` to select which click point
to use.

> === Configuration

Reviewer: I wonder why you put this section into "GettingStarted" and not "Installing". Especially because we now by default do not need files like openqa.ini and such anymore for default operation I think it's better suited to move this section to Installing and keep this documented tailored for users which might not even need to maintain their own openQA installation.

Author: I added it to "GettingStarted" because this document is the first and already contains mentionings of `/etc/openqa/openqa.ini`. So I thought it makes most sense to explain the concept of configuration upfront as it is a prerequisite for following sections.

I must also say that I find the configuration in general a very fundamental thing to know about when using software. It seems actually more important than some of the other topics we mention on "GettingStarted".

I can nevertheless move it to "Installing" if you like that better.

Reviewer: Open to hear feedback from others. As this is about documentation also "beginners" can provide helpful insight

Reviewer 2: Maybe @realcharmer could provide some useful feedback here?

Reviewer 3: Whilst looking at it as a relative newcomer, I prefer the way @Martchus structures it. I'd like to know what each configuration file does before diving into installation itself. I might also already be using an existing instance without having to install it myself from scratch. Having an initial overview of the config file is good in my opinion, although i do understand what @okurz is trying to point out.

> The different components of openQA read their configuration from the following
> files:
> 
> * `openqa.ini`: This is the "web UI" config as services providing the web
>   interface and related services such as the scheduler are configured via that
>   file.

Reviewer: This makes it sounds like this file would actually be necessary but by default we would not even need it anymore. We could try to start with the  "Locations" section and then put this list together with drop-in configurations but I guess you wanted to explain first configurations applicable for the individual components. How about leaving out the extension and not taking about files yet but call it "configuration components"? Something like

```suggestion
The different components of openQA use the following configuration components:

* `openqa`: This is the "web UI" configuration for services providing the web
  interface and related services such as the scheduler.
```

Author: That's bad because there is no "component" that is literally called `openqa`. I would rather explicitly name these configuration files. When I read documentation I usually don't want any abstract bla-bla but concrete information.

I can add a note that the files are not required to exist.

Author: For now I only fixed the phrasing and added an explicit note stating that defaults are used if the files are not present.

  authenticate with the web interface. One API key/secret can be configured per
  web UI host.

If these files are not present, defaults are used.

Example configuration files are installed under `/usr/share/doc/openqa/examples`
and in the
https://github.com/os-autoinst/openQA/tree/master/etc/openqa[Git repository under `etc/openqa`].
Continue reading the next sections for where you can place the actual
configuration files and the possibility of creating drop-in configuration files.

==== Locations
All configuration files can be placed under `/etc/openqa`,
e.g. `/etc/openqa/openqa.ini`. That is where administrators are expected to
store system-wide configuration.

Configuration files are also looked up under `/usr/etc/openqa` where a
https://en.opensuse.org/openSUSE:Packaging_UsrEtc#Variant_1_(ideal_case)[package maintainer]
can place default values deviating from upstream defaults.

The client configuration can also be put under `~/.config/openqa/client.conf`.

For development, checkout the section about
<<Contributing.asciidoc#_customize_configuration_directory,customizing the configuration directory>>.

==== Drop-in configurations

> It is possible to split the configuration into multiple files and store these

Reviewer: Can we emphasize that drop-in configurations are preferred to the complete overrides? I am worried because now you *first* describe the complete ini files and then mention drop-in that people assume they would need or should use openqa.ini. WDYT?

Reviewer 2: I would lean towards just mentioning the legacy config files which we continue to support at the end. That's what they are to my mind.

you must pass it to `testapi::assert_and_click` to select which click point
to use.

=== Configuration
The different components of openQA read their configuration from the following
files:

* `openqa.ini`, `openqa.ini.d/*.ini`: These files are the "web UI" config.
  Services providing the web interface and related services such as the

>   scheduler are configured via these files.

Reviewer 4: no problem for now but I found it too generic. And later it says `If these files are not present, defaults are used.` but which is the default if files appears in more than one path
